### PR TITLE
Request body stream

### PR DESCRIPTION
**Pull request by alexjeffburke**
This adds support for specifying the body of a request as a stream.

### CODE REVIEW

File: lib/unexpectedHttp.js [hunk 1]
```diff
@@ -4,6 +4,10 @@ var http = require('http'),
     messy = require('messy'),
     _ = require('underscore');
 
+function isStream(thing) {
+    return typeof thing.pipe === 'function';
```
Comment (papandreou) [inline lib/unexpectedHttp.js]: Breaks for `null` and `undefined`.

**Comment by papandreou** [comment]
:+1:

**Comment by papandreou** [comment]
Added you as an owner, feel free to merge whenever:)

File: lib/unexpectedHttp.js (hunk 1 above)
Comment (alexjeffburke) [inline lib/unexpectedHttp.js]: Yeah, was in two minds about that too - where it's used the null/undefined check was already done by the surrounding if, but it's not exactly a general test. I'm actually inclined just to move the `typeof <...>.pipe` bit inline wth a comment it's a check for a stream

**Pull request merged**